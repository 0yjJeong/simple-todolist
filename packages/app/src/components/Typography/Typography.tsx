import { FC } from 'react';
import styled from 'styled-components';

export enum Size {
  TITLE,
  SUB_TITLE,
  CAPTION,
}

type Styles = {
  '@size': number;
  '@weight': string;
};

export const typoStyles: {
  [key in Size]: Styles;
} = {
  0: {
    '@size': 36,
    '@weight': '500',
  },
  1: {
    '@size': 24,
    '@weight': '500',
  },
  2: {
    '@size': 12,
    '@weight': '500',
  },
};

export type TypographyProps = {
  text: string;
  size: Size;
};

export const TypographyBlock = styled.p<Styles>`
  margin: 0;
  font-size: ${(props) => props['@size']}px;
  font-weight: ${(props) => props['@weight']};
`;

const Typography: FC<TypographyProps> = ({ text, size }) => {
  return <TypographyBlock {...typoStyles[size]}>{text}</TypographyBlock>;
};

export default Typography;
