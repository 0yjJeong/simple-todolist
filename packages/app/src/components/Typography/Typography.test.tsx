import { render } from '@testing-library/react';
import 'jest-styled-components';
import Typography, { Size, typoStyles } from './Typography';

describe('<Typography/>', () => {
  it('should render title typo', () => {
    const styles = typoStyles[Size.TITLE];
    const rendered = render(<Typography text='a' size={Size.TITLE} />);
    expect(rendered.container.firstChild).toMatchSnapshot();
    expect(rendered.container.firstChild).toHaveStyleRule(
      'font-size',
      `${styles['@size']}px`
    );
    expect(rendered.container.firstChild).toHaveStyleRule(
      'font-weight',
      styles['@weight']
    );
  });

  it('should render sub-title typo', () => {
    const styles = typoStyles[Size.SUB_TITLE];
    const rendered = render(<Typography text='a' size={Size.SUB_TITLE} />);
    expect(rendered.container.firstChild).toMatchSnapshot();
    expect(rendered.container.firstChild).toHaveStyleRule(
      'font-size',
      `${styles['@size']}px`
    );
    expect(rendered.container.firstChild).toHaveStyleRule(
      'font-weight',
      styles['@weight']
    );
  });

  it('should render caption typo', () => {
    const styles = typoStyles[Size.CAPTION];
    const rendered = render(<Typography text='a' size={Size.CAPTION} />);
    expect(rendered.container.firstChild).toMatchSnapshot();
    expect(rendered.container.firstChild).toHaveStyleRule(
      'font-size',
      `${styles['@size']}px`
    );
    expect(rendered.container.firstChild).toHaveStyleRule(
      'font-weight',
      styles['@weight']
    );
  });
});
