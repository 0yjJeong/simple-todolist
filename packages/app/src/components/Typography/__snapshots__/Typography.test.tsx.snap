// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Typography/> should render caption typo 1`] = `
.c0 {
  margin: 0;
  font-size: 12px;
  font-weight: 500;
}

<p
  class="c0"
>
  a
</p>
`;

exports[`<Typography/> should render sub-title typo 1`] = `
.c0 {
  margin: 0;
  font-size: 24px;
  font-weight: 500;
}

<p
  class="c0"
>
  a
</p>
`;

exports[`<Typography/> should render title typo 1`] = `
.c0 {
  margin: 0;
  font-size: 36px;
  font-weight: 500;
}

<p
  class="c0"
>
  a
</p>
`;
